ref-exchange/src/account_deposit.rs

```diff
@@ -43,11 +91,23 @@ impl Account {
         self.tokens.insert(token.clone(), value - amount);
     }
 
+    /// Updates the account atorage usage,
+    /// Panic if there is not enough $NEAR to cover.
+    pub fn update_storage(&mut self, tx_start_storage: StorageUsage) {
+        // note: can not use += , 
+        // or a pannic with `attempt to subtract with overflow` emerges,
+        // if `tx_start_storage` is larger than `env::storage_usage()`
+        self.storage_used = self.storage_used + env::storage_usage() - tx_start_storage;
+        self.assert_storage_usage();
+    }
+
     // [AUDIT_01]
     /// Returns amount of $NEAR necessary to cover storage used by this data structure.
     pub fn storage_usage(&self) -> Balance {
-        (MIN_ACCOUNT_DEPOSIT_LENGTH + self.tokens.len() as u128 * (MAX_ACCOUNT_BYTES + 16))
-            * env::storage_byte_cost()
+        self.storage_used as u128 * env::storage_byte_cost()
```

Comment:
either storage used should be initialized to `(MIN_ACCOUNT_DEPOSIT_LENGTH + self.tokens.len() as u128 * (MAX_ACCOUNT_BYTES + 16))` or should still be adding.

e..g https://github.com/ref-finance/ref-contracts/blob/ref-components/ref-components/src/account_impl.rs#L68